TOSegmentedControl/TOSegmentedControl.m

```diff
@@ -1091,6 +1091,39 @@ - (BOOL)selectedSegmentReversed
     return segment.isReversed;
 }
 
+
+// -----------------------------------------------
+// Manually change thumb horizontal offset to connect it with UIPageController scroll
+
+- (void)setThumbHorizontalOffset:(CGFloat)offsetPercent direction:(NSInteger)direction
+{
+    //Get current thumb frame
+    CGRect frame = [self frameForSegmentAtIndex:self.selectedSegmentIndex];
+    
+    //Variable that will hold new X position for thumb
+    __block CGFloat newX = 0;
+    
+    //Animation block to move thumb
+    id animationBlock = ^{
+        if (direction > 0) {
+            newX = frame.origin.x + frame.size.width*offsetPercent;
+        } else {
+            newX = frame.origin.x - frame.size.width*offsetPercent;
+        }
+        [self.thumbView setFrame:CGRectMake(newX, frame.origin.y, frame.size.width, frame.size.height)];
+    };
+    
+    //Commit the animation
+    [UIView animateWithDuration:0.45
```

Comment:
Haha, I'm guessing you just copy and pasted my code here, but it's not necessary to animate the X offset. If you're updating the position on each tick of the scroll view, there's no need to do an animation.